### DISCUSSION TITLE
UI doesn't trigger right after checkForUpdatesInBackground() finishes

### DISCUSSION THREAD

**Discussion by albrt-dev** [sparkle-project/Sparkle#2567] [Q&A]
My application has silent checking and downloading updates through calling checkForUpdatesInBackground().

In Sparkle 1.x there was method installUpdatesIfAvailable() in SUUpdater which showed UI after call checkForUpdatesInBackground().
Now in Sparkle 2.x I have SPUUpdater with methods: checkForUpdates() (it shows UI - ok) and checkForUpdatesInBackground().

My question is how I can call UI after checkForUpdatesInBackground()?

P.S. There is a hack in calling checkForUpdates() after checkForUpdatesInBackground() finishes its work, but UI shows after 20-30 seconds, which unsuitable for me.

**Comment by zorgiepoo** [comment]
If you have automatic downloading of updates enabled, the update will be installed silently when the app terminates, so users do not need to be prompted (the best silent update is updating the users don't notice). However Sparkle does prompt if the update hasn't still been installed after the app has been running for 1 week, or prompts sooner if user authorization is required to install the update (because that will require user interaction). There are some further things can be customized here.. like Sparkle provides a callback to relaunch/install the update once it has been set to automatically install on quit.

If you still want to show users you are downloading/installing/relaunching an update on app launch (not recommended), you may want to consider implementing a [custom user interface](https://sparkle-project.org/documentation/api-reference/Protocols/SPUUserDriver.html) and making sure to disable automatic downloading of updates. Downloading an update can take time so you don't want to show a prompt 10-30 seconds later after downloading the update in the background.
